src/data_parser.py

```diff
@@ -1,20 +1,47 @@
 import data_fetcher
+import json
 
 
 def get_subjects(year: int):
+    """Return a list of subjects for a given year """
+    
     subjects = data_fetcher.DataFetcher(
         f"https://courseplanner-api.adelaide.edu.au/api/course-planner-query/v1/?target=/system/SUBJECTS_BY_YEAR/queryx&virtual=Y&year_from={year}&year_to={year}"
     )
-    res = subjects.get()
-
-    data = []
-    if res == {}:
-        return data
+    data = subjects.get()["data"]
+    
+    subject_list = []
+    for subject in data:
+        subject_list.append(subject["SUBJECT"])
+        
+    return {"subjects": subject_list} 
+   
+def get_course_ids(subject_code: int, year: int):
+    """Return a list of course ids for a given subject code and year"""
+    
+    courses = data_fetcher.DataFetcher(
+        f"https://courseplanner-api.adelaide.edu.au/api/course-planner-query/v1/?target=/system/COURSE_SEARCH/queryx&virtual=Y&year={year}&pagenbr=1&pagesize=500&subject={subject_code}"
+    )
+    data = courses.get()["data"]
+    
+    course_ids = []
+    for course in data:
```

Comment:
If the `get()` function returned an empty dict (in the case of an API request error) then the "data" field will be `None`. Trying to iterate over `None` will result in errors so you'll need to do some checking to avoid bugs

Reply:
I did not implement any exception handling for when the API call fails, because I feel like the most effective thing to do is to retry the request after waiting a few seconds, which should be implemented when the rate limiter gets implemented.

Reply:
Yeah that should be fine. I was mainly referring to checking if `data` was `None`, which it seems like you've implemented with

```py
if data is None:
    return {"courses": []}
```